### PR TITLE
Reference the book Haskell Programming from First Principals

### PR DESCRIPTION
**Pull request by jasonkuhrt**

### CODE REVIEW

**Comment by Gabriella439** [comment]
So I'm actually aware of that book, but I didn't mention it mainly because it was not complete yet.  My policy for this summary was to prefer things that were already completed and ready for people to use.

I'll leave this pull request open, though, and once the book comes out I will study it more closely and see if it fills in the remaining gaps.

**Comment by jasonkuhrt** [comment]
For what its worth there is an early release available that has hundreds of pages of content already. Its amazing so far. But yep seems fine, this can be revisited later once there are no such caveats. Thanks!

**Comment by GetContented** [comment]
@jasonkuhrt why amazing?

**Comment by jasonkuhrt** [comment]
Hi @GetContented, I've had a chance since creating this issue to progress over half way through the book. Here are some reasons:
- Carefully ordered content that consistently nails knowledge dependencies. For example the first chapter is on Lamba Calculus.
- Exercises both between sections and at the end of chapters that are already pretty good and consistently effective at reinforcing the content under review. However this is a part of the book that has been receiving more love as new releases come out. I've mentioned to them that venturing into more domains with the exercises can be helpful to really testing knowledge; I also suggested to append harder questions. Not all questions have answers yet. Still, the present questions are already invaluable.
- Unambiguous. Terminology is well introduced and applied consistently. Ideas are shown from multiple points of view. Generally, by the end of a chapter, they have been hammered in over and over the idea(s) at different angles such that you cannot be confused. The crescendo of exercises guarantees this.
- Thorough. The book starts with a brief introduction to Lamba Calculus (already mentioned) so that the underlying reason for currying can be understood and helps give the reader an impressive appreciation/intuition for functions and purity. This makes sense. The reader needs to fall in love with functions to succeed with Haskell I think. I've suggested a similar abstract chapter on Category Theory before going into Types. They received the same request from other readers and said they are considering potentially doing something.
- If Twitter is any indication reader approval has been very positive.
- Exercises. Repeating this again. They are _so_ important, and I wouldn't be succeeding the way I am now without them.

**Comment by GetContented** [comment]
Yeah, I'm actually now incredibly familiar with the book (being one of its reviewers... and having talked to Chris a lot about it, and also been one of the people who gave a review of it on the feedback page of the website).

I think it's definitely already at a point where it should be recommended on the SOTU, because it's already well better than the other Haskell beginner offerings (not complete programming language beginners though, probably).

Interesting that you highlight the exercises. I was recommending that there be many more of them, and that their content be more engaging... :) easier to critique something than to make something oneself though, so... yes, it's excellent.

**Comment by jasonkuhrt** [comment]
> I was recommending that there be many more of them, and that their content be more engaging

I would second this! While I think that they already are a huge asset to the book (e.g. crucial to making the content "stick" long-term), they can still be meaningfully better both in small and big ways.

**Comment by Gabriella439** [comment]
Yeah, I will add Haskell Book to the section.  I'm a little bit busy at the moment with Twitter's Hack week but I will try to merge this in soon

**Comment by jasonkuhrt** [comment]
@Gabriel439 :+1: No rush.

**Comment by Gabriella439** [comment]
Could you merge `master` into this pull request so that I can merge it in?

**Comment by jasonkuhrt** [comment]
@Gabriel439 Done.

File: sotu.md [hunk 1]
```diff
@@ -1285,8 +1285,6 @@ Haskell has decent logging support.  That's pretty much all there is to say.
 
 <br>
 
-## Education
```
Comment (Gabriella439) [inline sotu.md:1288]: Did you mean to delete this?

Comment (jasonkuhrt) [inline sotu.md:1288]: Hm, don't think so! Fixing.

Comment (jasonkuhrt) [inline sotu.md:1288]: Fixed.

**Comment by Gabriella439** [comment]
Thanks! :)

**Pull request merged**